### ISSUE TITLE
Forcing pipeline to use only on agent ?

### ISSUE THREAD

**Issue by alphaxr6** [gocd/azure-elastic-agent-plugin#53]
Hello,

Is there a way to force a pipeline to use the same agent for all the stages and the jobs ? 

Thank you

**Comment by chadlwilson** [comment]
Do you mean to reuse the same agent instance or just require a pipeline to use the same agent profile across all its jobs?

**Comment by alphaxr6** [comment]
My pipeline already use the same agent profile. I guess I meant to reuse the same agent instance for the pipeline

**Comment by chadlwilson** [comment]
What you are trying to achieve?

Does configuring the agent idle timeout not achieve what you are looking for?

**Comment by alphaxr6** [comment]
I've setup a pipeline with 3 different stages:

- Preconfig stage
- Terraform stage
- Configuration stage

There is some terraform output that needs to be use into the configuration stage.

I would like my all pipeline (all the stages) to run on the same VM (agent). It happened very often that when I manually trigger my pipeline, the plugin deploy 2 VM with 2 agents.

So, it happened that: 

- Terraform stage run on agent A
- Configuration stage run on agent B

So the configuration stage doesn't retrieve the terraform output of the terraform stage as it was not run on the same agent.

I would like to know if there is a way to make sure all the stages of a specific pipeline run on the same VM (agent)

**Comment by joestr** [comment]
If you are in _Stage B_ and depend on artifacts from _Stage A_, _Stage A_ needs artifacts to be configured. _Stage B_ needs to fetch those artifacts so it does not matter on which instance its running.

**Comment by alphaxr6** [comment]
Ok so I need to export my output from stage A into artifacts to use it into stage B, I get it

**Comment by chadlwilson** [comment]
Yes, you should not design your jobs to depend on where they are run, relative to other jobs. You should use either GoCD's artifacts or some other artifact store.

Alternatively, consider whether these things actually need to be different stages/jobs instead of just sequential steps/tasks within the same job.

I often find people excessively use stages/jobs when they should just use tasks. There are pros to splitting into stages, but there are also cons.
